modules/web/js/ballerina/diagram2/views/default/components/decorators/panel-decorator.jsx

```diff
@@ -156,6 +169,25 @@ class PanelDecorator extends React.Component {
 
         staticButtons.push(React.createElement(PanelDecoratorButton, deleteButtonProps, null));
 
+        // Render the public/private toggle flag button only if its a function
+        if ((!TreeUtils.isMainFunction(this.props.model) && TreeUtils.isFunction(this.props.model)) ||
+            TreeUtils.isStruct(this.props.model) || TreeUtils.isConnector(this.props.model)) {
+            // Toggle button for public/private flag
+            const publicPrivateFlagButtonProps = {
+                bBox: {
+                    x: x - (width * 3),
+                    y,
+                    height,
+                    width,
+                },
+                icon: ImageUtil.getSVGIconString(this.props.model.public ? 'public' : 'lock'),
```

Comment:
Can `this.props.model.public` be undefined at anytime ?
If so, `this.props.model.public === false`  will also give 'lock' and `this.props.model.public === undefined` will also give 'lock' as output.

Reply:
this.props.model.public should be either true or false. It can never be undefined

Reply:
👍